lib/task_005.rb

```diff
@@ -48,52 +49,18 @@
 module Task005
   module_function
 
-  def parse_input(input_data)
-    input_data.split("\n").map { |line| parse_line(line) }
-  end
-
-  def parse_line(line)
-    point1, point2 = line.split(' -> ')
-
-    {
-      point1: parse_point(point1),
-      point2: parse_point(point2),
-    }
-  end
-
-  def parse_point(point)
-    point.split(',').map { |value| value.to_i }
-  end
-
-  def mark_lines(lines)
-    diagram = {}
-    lines.each { |line| mark_line(line, diagram) }
-    diagram
-  end
-
-  def mark_line(line, diagram)
-    if line[:point1][0] == line[:point2][0] || line[:point1][1] == line[:point2][1]
-      x1, x2 = [line[:point1][0], line[:point2][0]].sort
-      y1, y2 = [line[:point1][1], line[:point2][1]].sort
-      (x1..x2).each do |x|
-        (y1..y2).each do |y|
-          diagram.key?([x, y]) ? diagram[[x, y]] += 1 : diagram.store([x, y], 1)
-        end
-      end
-    end
-    diagram
-  end
-
   def calculate_answer(input_data)
     lines = parse_input(input_data)
-    diagram = mark_lines(lines)
-    diagram.count { |_key, value| value >= 2 }
+
+    diagram = Diagram.new
+    diagram.mark_horizontal_and_vertical(lines)
+    diagram.count_values_greater_than_2
```

Comment:
nice!

Reply:
I don't like word "values" as it is too generic. You could use "points".

Bye the way you could also pass `2` as an argument :D

```suggestion
    diagram.count_points_with_min_coverage(2)
```